stan/math/gpu/opencl_context.hpp

```diff
@@ -101,16 +103,32 @@ class opencl_context_base {
       check_opencl_error("opencl_context", e);
     }
   }
+
+  // Helper func for reading in kernels
+  inline const char* read_kernel(const char* file_name) {
+    std::ifstream kernel_buff(file_name);
+    static std::string kernel;
+    kernel_buff.seekg(0, std::ios::end);
+    kernel.reserve(kernel_buff.tellg());
+    kernel_buff.seekg(0, std::ios::beg);
+    kernel.assign((std::istreambuf_iterator<char>(kernel_buff)),
+                  std::istreambuf_iterator<char>());
+    return kernel.c_str();
```

Comment:
This return isn't safe. We ran into a problem with this years ago. See: https://stackoverflow.com/questions/22330250/how-to-return-a-stdstring-c-str

Mind checking if this is the `static` problem you saw?

Is it possible to operate using `std::string` where this is used?